### PR TITLE
[MGDOBR-208] - HTTP failures of the shard should be included in the m…

### PR DESCRIPTION
**Pull request by vaibhavjainwiz**
…etrics

[Insert JIRA Issue number and update link](https://issues.redhat.com/browse/MGDOBR-) 
https://issues.redhat.com/browse/MGDOBR-208

With this change new metrics will published whenever shard tries to fetch/update manager. Metrics such as below would be published.
```
# HELP http_manager_request_total  
# TYPE http_manager_request_total counter
http_manager_request_total{shardId="shard-1",status="FAILURE",type="UPDATE",} 6.0
http_manager_request_total{shardId="shard-1",status="SUCCESS",type="UPDATE",} 3.0
http_manager_request_total{shardId="shard-1",status="SUCCESS",type="FETCH",} 18.0
http_manager_request_total{shardId="shard-1",status="FAILURE",type="FETCH",} 2.0
```
Please make sure that your PR meets the following requirements:

- [x] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config)
- [x] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [x] All new functionality is tested
- [x] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

* <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

* <b>Rebase the pull request</b>  
  Comment with: `/rebase`.

</details>

### CODE REVIEW

File: shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/metrics/MetricsService.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package com.redhat.service.bridge.shard.operator.metrics;
+
+public interface MetricsService {
+    void UpdateManagerRequestMetrics(ManagerRequestType requestType, ManagerRequestStatus status);
```
Comment (r00ta) [inline shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/metrics/MetricsService.java:4]: small typo `UpdateManagerRequestMetrics` -> `updateManagerRequestMetrics`

File: shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/metrics/MetricsServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    @Inject
+    MeterRegistry meterRegistry;
+
+    @ConfigProperty(name = "event-bridge.sso.grant-options.password.username")
+    String username;
+
+    @Override
+    public void UpdateManagerRequestMetrics(ManagerRequestType requestType, ManagerRequestStatus status) {
+        List<Tag> tags = new ArrayList<>();
+        tags.add(Tag.of("shardId", username));
```
Comment (r00ta) [inline shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/metrics/MetricsServiceImpl.java:28]: `event-bridge.sso.grant-options.password.username` is not really the `shardId`. You have to inject it manually, for example the `shardId` of the `shard-1` user is https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/3b0597ebdd886d15035f3faee205f4a465e5f2f2/kustomize/overlays/minikube/keycloak/event-bridge-fm-realm-sample.json#L43 . 
You might create a new config property like `event-bridge.shard-id` and configure it accordingly

Comment (vaibhavjainwiz) [reply shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/metrics/MetricsServiceImpl.java:28]: @r00ta `username` is more readable. Could we keep it to `username` and change tag name to `shardName`.
WDYT?

File: shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/metrics/MetricsService.java (hunk 1 above)
Comment (vaibhavjainwiz) [reply shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/metrics/MetricsService.java:4]: done

File: shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/metrics/MetricsServiceImpl.java (hunk 2 above)
Comment (vaibhavjainwiz) [reply shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/metrics/MetricsServiceImpl.java]: changed shardName to shardId

File: shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/metrics/MetricsServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    private static final String MANAGER_REQUEST_METRICS = "http.manager.request";
+
+    @Inject
+    MeterRegistry meterRegistry;
+
+    @ConfigProperty(name = "event-bridge.sso.grant-options.password.user-id")
+    String shardId;
+
+    @Override
+    public void updateManagerRequestMetrics(ManagerRequestType requestType, ManagerRequestStatus status) {
```
Comment (robpblake) [inline shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/metrics/MetricsServiceImpl.java:26]: @vaibhavjainwiz Is it also useful to include the HTTP status received when trying to call the Manager here as a tag?

Comment (robpblake) [inline shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/metrics/MetricsServiceImpl.java:26]: @vaibhavjainwiz Is it also useful to include the HTTP status received when trying to call the Manager here as a tag?

File: shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/ManagerSyncServiceImpl.java [hunk 4]
```diff
@@ -64,19 +70,25 @@ void syncUpdatesFromManager() {
     public Uni<HttpResponse<Buffer>> notifyBridgeStatusChange(BridgeDTO bridgeDTO) {
         LOGGER.debug("Notifying manager about the new status of the Bridge '{}'", bridgeDTO.getId());
         return getAuthenticatedRequest(webClientManager.put(APIConstants.SHARD_API_BASE_PATH), request -> request.sendJson(bridgeDTO))
+                .onItem().invoke(success -> metricsService.updateManagerRequestMetrics(ManagerRequestType.UPDATE, ManagerRequestStatus.SUCCESS))
+                .onFailure().invoke(failure -> metricsService.updateManagerRequestMetrics(ManagerRequestType.UPDATE, ManagerRequestStatus.FAILURE))
                 .onFailure().retry().withBackOff(WebClientUtils.DEFAULT_BACKOFF).withJitter(WebClientUtils.DEFAULT_JITTER).atMost(WebClientUtils.MAX_RETRIES);
```
Comment (robpblake) [inline shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/ManagerSyncServiceImpl.java:75]: @vaibhavjainwiz I find this quite hard to follow. The second `onFailure` call, what is going to be retried? The original  call the Fleet Manager, or the attempt to update the `metricsService`?

Comment (robpblake) [inline shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/ManagerSyncServiceImpl.java:75]: @vaibhavjainwiz I find this quite hard to follow. The second `onFailure` call, what is going to be retried? The original  call the Fleet Manager, or the attempt to update the `metricsService`?

File: shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/metrics/MetricsServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+        List<Tag> tags = new ArrayList<>();
+        tags.add(Tag.of("shardId", shardId));
```
Comment (robpblake) [inline shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/metrics/MetricsServiceImpl.java:28]: @vaibhavjainwiz These can be constants, the same way you've defined `MANAGER_REQUEST_METRICS`.

Comment (robpblake) [inline shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/metrics/MetricsServiceImpl.java:28]: @vaibhavjainwiz These can be constants, the same way you've defined `MANAGER_REQUEST_METRICS`.

File: shard-operator/src/main/resources/application.properties [hunk 6]
```diff
@@ -18,6 +18,7 @@ event-bridge.sso.auth-server-url=${EVENT_BRIDGE_SSO_URL:http://${minikubeip}:300
 event-bridge.sso.client-id=${EVENT_BRIDGE_SSO_CLIENT_ID:event-bridge}
 event-bridge.sso.credentials.secret=${EVENT_BRIDGE_SSO_SECRET:secret}
 event-bridge.sso.grant.type=password
+event-bridge.sso.grant-options.password.user-id=${KEYCLOAK_USER_ID:150b23e9-0b34-45b7-a91a-5ec388d03a1d}
```
Comment (robpblake) [inline shard-operator/src/main/resources/application.properties:21]: Feels like this would be better named `event-bridge.shard-id`?

File: kustomize/overlays/minikube/shard/kustomization.yaml [hunk 7]
```diff
@@ -1,6 +1,7 @@
 secretGenerator:
```
Comment (robpblake) [inline kustomize/overlays/minikube/shard/kustomization.yaml:1]: @vaibhavjainwiz We'll also need any required variables added to our `prod` and `stable` overlays as well. There are instructions in the `kustomize/README.md` on how to do that.

cheers,

Rob

**Review by robpblake — COMMENTED** [review]
Thanks @vaibhavjainwiz . Some comment from me.

cheers,
Rob

File: shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/ManagerSyncServiceImpl.java (hunk 4 above)
Comment (vaibhavjainwiz) [reply shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/ManagerSyncServiceImpl.java:75]: The original call of Fleet Manager will be retry again on failure. Metrics will be published with every failed attempt for call to fleet manager.

Comment (vaibhavjainwiz) [reply shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/ManagerSyncServiceImpl.java:75]: The original call of Fleet Manager will be retry again on failure. Metrics will be published with every failed attempt for call to fleet manager.

File: shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/metrics/MetricsServiceImpl.java (hunk 3 above)
Comment (vaibhavjainwiz) [reply shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/metrics/MetricsServiceImpl.java:26]: done

File: shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/metrics/MetricsServiceImpl.java (hunk 5 above)
Comment (vaibhavjainwiz) [reply shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/metrics/MetricsServiceImpl.java:28]: done

File: kustomize/overlays/stable/secrets/keycloak-secrets.yaml [hunk 8]
```diff
@@ -7,6 +7,7 @@ metadata:
   name: keycloak-secrets
 spec:
   encryptedData:
+    KEYCLOAK_USER_ID: AgBqr9S/CBggW4OIXgo8wRuF9CpuOsIxPNqaure6XztYmGqNlTTQKrSw0gZQgjy7rDWkffeC8q/nwXsPhaQLU9qaEGzu9mBnQUi7N7aq06QpNx4HLXmDYfhwB97ULqVViRy9tmStVknlL3CAaT/K8oXVoqAcOSB5sSeLDH5eCIyf4iW7PfpOSPK5zZPXoRahd15r/YqGOzLjSsC8KRise6o7oDYdIcBL0GdvrslXuCEF5s9lsMVRe2ZTmKQg9OD2WG+jdgRlS4ryUnhWDgWSpHTo9iCL6vY+kUZ/lTC7gBSxoplxaYWeOGgwfeooFI2Fc2euqi3m8JmKlAQ2FOkC8ij84gHlv7sCoTTA/+HVQwd8yiIDmqFyjneswbWSGTzcjy4motoSnrFxvKYVo3BfO6ypYxMt3sh7T1TtQGgyqOLIyNJlNF5lcDDAIalOwFLure0nDy9yF4SQ/q2BXAGZM/3HxVjdBb/SgJcStJ3/a+3fh0aeN9X3qlhyJLtTqc616cgQEz2BuV9bYfW3Eq5hNPjcteVaXbA6lzZZgFGof6hpW/NK14n3BThMJfvRHc1hwrxvhzRK9kirzra5X3Wuw5TZHNo3pNtPEm+ybqDesjtqEHLl8k2dR4S1pndGTY6Uq5zETOZZjNYyGoEVdqMfkBKQUXJ6SiwOm7pZEv2dDXaaQEPDUCaH3gui1JSCrDedojP/xZ+mU/eJ0L/uXO4ZrIFqxXYleMdvOaE/vYEMRHjkTccjHq0=
```
Comment (kostola) [inline kustomize/overlays/stable/secrets/keycloak-secrets.yaml:10]: This string is exactly the same as the one in the `prod` overlay, but it can't work this way because the two environments use different encryption keys.
You must regenerate the secret using the `stable` encryption key instead of copy the one from `prod`.

Comment (robpblake) [reply kustomize/overlays/stable/secrets/keycloak-secrets.yaml:10]: @kostola The update to our `kustomize/README.md` is part of https://issues.redhat.com/browse/MGDOBR-331, which is assigned to you. Can you do an update to the README.md to provide the instructions on which key to use so that the rest of the team is aware of the need to do that.

cheers,

Rob

Comment (vaibhavjainwiz) [reply kustomize/overlays/stable/secrets/keycloak-secrets.yaml]: Updated `stable` encryption key

File: shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/metrics/MetricsServiceImpl.java [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+    private static final String MANAGER_REQUEST_METRICS = "http.manager.request";
+    private static final String SHARD_ID = "shardId";
+    private static final String REQUEST_TYPE = "type";
+    private static final String REQUEST_STATUS = "status";
+    private static final String HTTP_STATUS_CODE = "statusCode";
+
+    @Inject
+    MeterRegistry meterRegistry;
+
+    @ConfigProperty(name = "event-bridge.sso.grant-options.password.shard-id")
```
[the remark this answers is not in the document]
Comment (r00ta) [reply shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/metrics/MetricsServiceImpl.java:26]: @kostola this class is in the shard itself, and this property is used to include the `shard-id` in the metrics that this particular instance will export. There is no jwt token involved in this, the instance needs to know what is its `shard-id` with a config property

[the remark this answers is not in the document]
Comment (vaibhavjainwiz) [reply shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/metrics/MetricsServiceImpl.java:26]: @r00ta With your changes in MGDOBR-486 you had removed `KEYCLOAK_USER` and `KEYCLOAK_PASSWORD` properties and move authencation to service account only. 
Currently I am fetching `shard_id` as `KEYCLOAK_USER_ID` which I think is no more relevant now([link](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/b22a472d3c3d78842a66f81b718e3609e4320742/shard-operator/src/main/resources/application.properties#L21)). How should I fetch `shard_id` now?

[the remark this answers is not in the document]
Comment (vaibhavjainwiz) [reply shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/metrics/MetricsServiceImpl.java]: Thanks @r00ta for detailed description. I had incorporated suggested remarks.

File: kustomize/overlays/prod/secrets/event-bridge-shard-operator-secrets.yaml [hunk 10]
```diff
@@ -11,6 +11,7 @@ spec:
...
+    EVENT_BRIDGE_SHARD_ID: AgBqr9S/CBggW4OIXgo8wRuF9CpuOsIxPNqaure6XztYmGqNlTTQKrSw0gZQgjy7rDWkffeC8q/nwXsPhaQLU9qaEGzu9mBnQUi7N7aq06QpNx4HLXmDYfhwB97ULqVViRy9tmStVknlL3CAaT/K8oXVoqAcOSB5sSeLDH5eCIyf4iW7PfpOSPK5zZPXoRahd15r/YqGOzLjSsC8KRise6o7oDYdIcBL0GdvrslXuCEF5s9lsMVRe2ZTmKQg9OD2WG+jdgRlS4ryUnhWDgWSpHTo9iCL6vY+kUZ/lTC7gBSxoplxaYWeOGgwfeooFI2Fc2euqi3m8JmKlAQ2FOkC8ij84gHlv7sCoTTA/+HVQwd8yiIDmqFyjneswbWSGTzcjy4motoSnrFxvKYVo3BfO6ypYxMt3sh7T1TtQGgyqOLIyNJlNF5lcDDAIalOwFLure0nDy9yF4SQ/q2BXAGZM/3HxVjdBb/SgJcStJ3/a+3fh0aeN9X3qlhyJLtTqc616cgQEz2BuV9bYfW3Eq5hNPjcteVaXbA6lzZZgFGof6hpW/NK14n3BThMJfvRHc1hwrxvhzRK9kirzra5X3Wuw5TZHNo3pNtPEm+ybqDesjtqEHLl8k2dR4S1pndGTY6Uq5zETOZZjNYyGoEVdqMfkBKQUXJ6SiwOm7pZEv2dDXaaQEPDUCaH3gui1JSCrDedojP/xZ+mU/eJ0L/uXO4ZrIFqxXYleMdvOaE/vYEMRHjkTccjHq0=
```
Comment (r00ta) [inline kustomize/overlays/prod/secrets/event-bridge-shard-operator-secrets.yaml:14]: This is outdated. `dev` is now using sso.redhat so you have to encrypt the new shard-id. Ping me on PM to get the new value to encrypt

File: kustomize/overlays/stable/secrets/event-bridge-shard-operator-secrets.yaml [hunk 11]
```diff
@@ -11,6 +11,7 @@ spec:
...
+    EVENT_BRIDGE_SHARD_ID: AgAJQooTOSC0pE1EjIdmtPXkJICTWrZiZCxysX2qaEFdAF5Af2qpmwZeo1PzI7uXW1Yn9nzpr8hUyLrcXfNyHA6p4A2Os2J/jE90COt2bgW45oLNA0iG6roWqK7CQIK+63WqMPPXch0mc0IPu5BNFF6vPf4qCLxQxhJOe8+2UWrbnSo2ELICPHkeRDrfOXg4SN0imS7Pmjhx3lGi0vRP5HLCIHwKhzXDETojqUQ5o2Oh5IX0yA5HwfYZL52uhIDkSoSG53fWClTuCjyBnAGi8KJtYQA22StaWSGLb1bk9+WxplYwY1zR6kFqBXZq6VU5gRecKhP+nACQZ7x9NheuidlZ/B+R12TKs+pEun/OS23hPQoQywDj6tsFM1IUkXujOaEV0ceD9cbAc4J7PZF9rYjQPWk/+qwWInaX+EnRKvjq1BP3ytgO7FJ3PookxKiYvAFMZCKx8aWqYhtXhETBP8CPR55pIYyLD9isZZvSgWb7hUDOVWFuyXVPguy2gE+BT2rm68cDQCPvMy8KnfrjdoPA4G5kZZxxVzWF1miVX+RX0CFMzVQGpgkhMntEj1tGdRkEvamteOY4J730pPO/hOqF8WKKKFvdhlSJYAENQMxt3UCbrahZdZ56B52F+aV4hanSeLFj9xg4Sv1ilFsxXUT7lj7Y6zRy8M1MzRaSOpHxjIZjULGhQnT6rBUH28qQMyt3mGGIehPZuYdaFjMp3KfY3vX3sFdV2e+m8a58aK25GROu8Pk=
```
Comment (r00ta) [inline kustomize/overlays/stable/secrets/event-bridge-shard-operator-secrets.yaml:14]: same here

File: openapi/openapi.yaml [hunk 12]
```diff
@@ -11,39 +11,39 @@ info:
     url: https://www.apache.org/licenses/LICENSE-2.0.html
   version: 0.0.1
 tags:
-- name: Bridges
-  description: "The API that allow the user to retrieve, create or delete Bridge instances."
-- name: Error Catalog
-  description: List and get the error definitions from the error catalog.
-- name: Processors
-  description: "The API that allow the user to retrieve, create or delete Processors\
+  - name: Bridges
```
Comment (r00ta) [inline openapi/openapi.yaml:14]: revert these changes

File: shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/ManagerSyncServiceImpl.java [hunk 13]
```diff
@@ -192,4 +207,16 @@ private void processingComplete(Class<?> entity) {
...
+        metricsService.updateManagerRequestMetrics(requestType, ManagerRequestStatus.SUCCESS, String.valueOf(successResponse.statusCode()));
+    }
+
+    private void updateManagerRequestMetricsOnFailure(ManagerRequestType requestType, Throwable error) {
+        if (error instanceof HTTPResponseException) {
+            int statusCode = ((HTTPResponseException) error).getStatusCode();
+            metricsService.updateManagerRequestMetrics(requestType, ManagerRequestStatus.FAILURE, String.valueOf(statusCode));
+        }
+        metricsService.updateManagerRequestMetrics(requestType, ManagerRequestStatus.FAILURE, StringUtils.EMPTY);
+    }
```
Comment (r00ta) [inline shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/ManagerSyncServiceImpl.java:221]: I think we should keep track of the failures also when they are not instanceof `HTTPResponseException`

File: shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/ManagerSyncServiceImpl.java [hunk 14]
```diff
@@ -192,4 +207,16 @@ private void processingComplete(Class<?> entity) {
     private boolean isSuccessfulResponse(HttpResponse<?> httpResponse) {
         return httpResponse.statusCode() >= 200 && httpResponse.statusCode() < 400;
     }
+
+    private void updateManagerRequestMetricsOnSuccess(ManagerRequestType requestType, HttpResponse<Buffer> successResponse) {
+        metricsService.updateManagerRequestMetrics(requestType, ManagerRequestStatus.SUCCESS, String.valueOf(successResponse.statusCode()));
+    }
+
+    private void updateManagerRequestMetricsOnFailure(ManagerRequestType requestType, Throwable error) {
+        if (error instanceof HTTPResponseException) {
```
Comment (robpblake) [inline shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/ManagerSyncServiceImpl.java:216]: @vaibhavjainwiz If the Error is a `HTTPResponseException`, we'll log the metric twice. Perhaps something like this instead:

```
String statusCode = null;
if(error instanceof HTTPResponseException) {
  statusCode = String.valueOf(((HTTPResponseException) error).getStatusCode());
}
metricsService.updateManagerRequestMetrics(requestType, ManagerRequestStatus.FAILURE, statusCode);

File: shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/ManagerSyncServiceImpl.java (hunk 13 above)
Comment (robpblake) [reply shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/ManagerSyncServiceImpl.java:221]: I think we do, we're just not extracting any HTTP Response code from them.

Comment (robpblake) [reply shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/ManagerSyncServiceImpl.java:221]: I think we do, we're just not extracting any HTTP Response code from them.

File: shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/metrics/MetricsServiceImpl.java [hunk 15]
```diff
@@ -0,0 +1,38 @@
...
+    @ConfigProperty(name = "event-bridge.shard-id")
+    String shardId;
+
+    @Override
+    public void updateManagerRequestMetrics(ManagerRequestType requestType, ManagerRequestStatus status, String statusCode) {
+        List<Tag> tags = new ArrayList<>();
+        tags.add(Tag.of(SHARD_ID, shardId));
+        tags.add(Tag.of(REQUEST_TYPE, requestType.name()));
+        tags.add(Tag.of(REQUEST_STATUS, status.name()));
+        tags.add(Tag.of(HTTP_STATUS_CODE, statusCode));
```
Comment (robpblake) [inline shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/metrics/MetricsServiceImpl.java:35]: Probably worth checking to see if the `statusCode` is null or empty? I think an empty statusCode tag is of no benefit.

File: kustomize/overlays/prod/secrets/event-bridge-shard-operator-secrets.yaml (hunk 10 above)
Comment (vaibhavjainwiz) [reply kustomize/overlays/prod/secrets/event-bridge-shard-operator-secrets.yaml:14]: done

File: kustomize/overlays/stable/secrets/event-bridge-shard-operator-secrets.yaml (hunk 11 above)
Comment (vaibhavjainwiz) [reply kustomize/overlays/stable/secrets/event-bridge-shard-operator-secrets.yaml:14]: done

File: shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/ManagerSyncServiceImpl.java (hunk 14 above)
Comment (vaibhavjainwiz) [reply shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/ManagerSyncServiceImpl.java:216]: sorry, I miss that condition. Thanks for pointing it out.

Comment (vaibhavjainwiz) [reply shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/ManagerSyncServiceImpl.java:216]: done

[the remark this answers is not in the document]
Comment (vaibhavjainwiz) [reply shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/ManagerSyncServiceImpl.java:216]: done

File: openapi/openapi.yaml (hunk 12 above)
Comment (vaibhavjainwiz) [reply openapi/openapi.yaml]: done

File: kustomize/overlays/minikube/shard/patches/deploy-config.yaml [hunk 16]
```diff
@@ -5,6 +5,7 @@ data:
   EVENT_BRIDGE_K8S_PLATFORM: k8s
   EVENT_BRIDGE_MANAGER_URL: http://event-bridge.event-bridge-manager.svc.cluster.local:8080
   EVENT_BRIDGE_SSO_URL: http://<REPLACE_WITH_MINIKUBE_IP>:30007/auth/realms/event-bridge-fm
+  EVENT_BRIDGE_SHARD_ID: "55095146"
```
Comment (r00ta) [inline kustomize/overlays/minikube/shard/patches/deploy-config.yaml:8]: `minikube` will use the local keycloak instance. Replace this with https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/ef4d9bb32b4ab05e1faf682a4664c09f48f4e3fd/kustomize/overlays/minikube/keycloak/event-bridge-fm-realm-sample.json#L43

Comment (vaibhavjainwiz) [reply kustomize/overlays/minikube/shard/patches/deploy-config.yaml]: done

File: shard-operator/src/main/resources/application.properties [hunk 17]
```diff
@@ -31,3 +31,4 @@ event-bridge.webhook.account-id=${WEBHOOK_TECHNICAL_ACCOUNT_ID:402cf429-da04-493
 %openshift.quarkus.kubernetes.deployment-target=openshift
 %openshift.quarkus.container-image.build=true
 %openshift.event-bridge.k8s.platform=openshift
+event-bridge.shard-id=${EVENT_BRIDGE_SHARD_ID:55095146}
```
Comment (r00ta) [inline shard-operator/src/main/resources/application.properties:34]: replace with 150b23e9-0b34-45b7-a91a-5ec388d03a1d

Comment (vaibhavjainwiz) [reply shard-operator/src/main/resources/application.properties]: done

**Review by r00ta — APPROVED** [review]

**Review by robpblake — APPROVED** [review]
@vaibhavjainwiz Thank you very much! Nice work.

cheers,

Rob
